src/Components/HomeCityCards/HomeCityCards.tsx

```diff
@@ -3,30 +3,33 @@ import './HomeCityCards.css'
 
 import { WeatherLocation, Forecast } from "../../model/Weather";
 import HomeCard from "../HomeCard/HomeCard"
+import { Link } from 'react-router-dom';
 
 interface HomeCityProps {
   allCities: WeatherLocation[];
   onSelect: (city: WeatherLocation) => void;
-  current: WeatherLocation | null;
+  clickedCard: WeatherLocation | null;
   details: Forecast | null;
```

Comment:
Going to need to add a <Loading /> component for the delay in fetching the details and setting that to state.